caddyhttp/httpserver/condition.go

```diff
@@ -123,7 +119,12 @@ func newIfCond(a, operator, b string) (ifCond, error) {
 		operator = operator[4:]
 	}
 	if _, ok := ifConditions[operator]; !ok {
-		return ifCond{}, operatorError(operator)
+		return ifCond{}, fmt.Errorf("Invalid operator %v", operator)
+	}
+	if operator == matchOp {
+		if _, err := regexp.Compile(b); err != nil {
```

Comment:
I tried doing that, ended up being quite painful. But I think I fixed the issue that put a stopper on it. I'll review my old stash and go from there.